lib/rubocop/cop/performance/string_replacement.rb

```diff
@@ -20,10 +20,15 @@ module Performance
       #   'a b c'.delete(' ')
       class StringReplacement < Cop
         MSG = 'Use `%s` instead of `%s`.'
-        DETERMINISTIC_REGEX = /^[\w\s\-,."']+$/
-        REGEXP_CONSTRUCTOR_METHODS = [:new, :compile]
-        GSUB_METHODS = [:gsub, :gsub!]
-        DETERMINISTIC_TYPES = [:regexp, :str, :send]
+        DETERMINISTIC_REGEX = /^[\w\s\-,."']+$/.freeze
```

Comment:
We should add a cop suggesting to freeze literal constants (and a rule to the style guide). This has been on my mind for a while now.